pkg/controller/clustercatalog_controller_test.go

```diff
@@ -0,0 +1,127 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestClusterCatalogControllerSync(t *testing.T) {
+	for _, tc := range []struct {
+		name        string
+		ctrl        *clusterCatalogController
+		shouldError bool
```

Comment:
I'd suggest changing this to the following:

```suggestion
		assertError func(t *testing.T, err error)
```

That way we don't actually assert on a specific error, it would be possible to accidentally introduce a different code path that errors in a different, unexpected way and for these tests to still pass.

Example of where I learned this lesson: https://github.com/operator-framework/operator-registry/pull/709

Reply:
Should be done in https://github.com/openshift/cluster-olm-operator/pull/73/commits/5b237d4e4d462f19ff7571661b839393bde6b676